public/pages/createDetector/components/DataFilters/SimpleFilter.tsx

```diff
@@ -76,6 +86,36 @@ export const SimpleFilter = (props: DataFilterProps) => {
   const lightModeStyles = {
     backgroundColor: '#F6F6F6',
   };
+
+  //If user search field name, filter filed names which include user's input word
+  //So user can only need to select from the filtered fileds.
+  const handleSearchFieldChange = debounce(async (searchValue: string) => {
+    let selectedFields: any = [];
+    if (searchValue) {
+      for (let i = 0; i < indexFields.length; i++) {
+        let selectedOptions: any = [];
+
+        let options = indexFields[i]?.options;
```

Comment:
I see the `?` is to prevent an undefined error, when would that occur in this case?

Reply:
Check this code https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/master/public/pages/createDetector/components/DataFilters/utils/helpers.ts#L33-L43